HW2/P4/P4.txt

```diff
@@ -0,0 +1,20 @@
+Number of threads = 1
+12.8246049881 seconds for 10 filter passes.
+
+Number of threads = 2
+7.46538496017 seconds for 10 filter passes.
+
+Number of threads = 4
+7.65428709984 seconds for 10 filter passes.
+
+
+Using two threads allows me to perform some operations in parallel and reduce the time of my operation by almost half.  The reduction is not quite half because there are communication costs and the threads have to wait for each other.  
+
+Consistent with my results in other parts of the Problem set, I see a performance enhancement from using 2 threads, but not 4, because the Virtual Machine only has 2 CPUs.  
+
```

Comment:
Good explanation. I never think of less enhancement is due to the number of CPUs.